llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -407,6 +450,29 @@ async def _run_workflow() -> None:
         asyncio.create_task(_run_workflow())
         return result
 
+    @dispatcher.span
+    def run_from(
+        self,
+        checkpoint: Checkpoint,
+        ctx_serializer: Optional[BaseSerializer] = None,
+        checkpoint_callback: Optional[CheckpointCallback] = None,
```

Comment:
Whats the intended purpose for the checkpoint callback? I think it got missed in the examples and I'm not 100% sure

Reply:
gotcha makes sense. Is this something the user would ever need to customize?

Reply:
I think the user could customize it in the future, if they wanted to inject some logic before checkpointing, such as persisting to some external storage.

Related, I think the user may want to be able to specify within a step a checkpoint, in which case I guess we would need to allow the user to include a `checkpoint_callback` in the step function definition.